editor/state.js

```diff
@@ -31,9 +31,37 @@ export const blocks = combineReducers( {
 		return state;
 	},
 	order( state = [], action ) {
+		let index;
+		let swappedUid;
 		switch ( action.type ) {
 			case 'REPLACE_BLOCKS':
 				return action.blockNodes.map( ( { uid } ) => uid );
+
+			case 'MOVE_BLOCK_UP':
```

Comment:
I have a small preference for the explicit actions but I don't mind if we change this later.